pkg/executor/types.go

```diff
@@ -74,27 +76,28 @@ func (j *Job) Copy() Job {
 // JobSpec is a complete specification of a job that can be run on some
 // execution provider.
 type JobSpec struct {
+	ApiVersion ApiVersion `json:"apiVersion,omitempty" yaml:"apiVersion,omitempty"`
 	// e.g. docker or language
-	Engine EngineType `json:"engine"`
+	Engine EngineType `json:"engine" yaml:"engine"`
 
 	// e.g. ipfs or localfs
 	// these verifiers both just copy the results
 	// and don't do any verification
-	Verifier verifier.VerifierType `json:"verifier"`
+	Verifier verifier.VerifierType `json:"verifier" yaml:"verifier"`
 
 	// executor specific data
-	Docker   JobSpecDocker   `json:"job_spec_docker"`
-	Language JobSpecLanguage `json:"job_spec_language"`
+	Docker   JobSpecDocker   `json:"job_spec_docker" yaml:"job_spec_docker"`
```

Comment:
the omitempty should go here (not below), as the whole part is optional - for both Docker and Language